static/js/publisher/market/market.js

```diff
@@ -88,32 +105,47 @@ function initForm(config, initialState, errors) {
   initSnapScreenshotsEdit(
     config.screenshotsToolbar,
     config.screenshotsWrapper,
-    state
+    state,
+    (nextState) => {
+      updateState(state, nextState);
+      updateFormState();
+    }
   );
 
-  // when anything is changed update the state
-  marketForm.addEventListener('change', function(event) {
-    let formData = new FormData(marketForm);
+  function checkForm() {
+    const diff = diffState(initialState, state);
+
+    if (diff && isFormValid()) {
+      enableSubmit();
+    } else {
+      disableSubmit();
+    }
+  }
 
+  function updateFormState() {
     // Some extra modifications need to happen for the checkboxes
-    publicMetrics(marketForm, formData);
+    publicMetrics(marketForm);
+
+    let formData = new FormData(marketForm);
+
+    // update state based on data of all inputs
     updateState(state, formData);
+    // checkboxes are tricky,
+    // make sure to update state based on their 'checked' status
+    updateState(state, {
```

Comment:
Making sure checkboxes are stored as booleans in the state (instead of "true" string from FormData).